### PR TITLE
pool: Make endpoint tests more robust and fix race.

### PR DESCRIPTION
**Pull request by davecgh**
This modifies the endpoint tests to respect the context when sending to and receiving from the endpoint channels similar to what was recently done for the chainstate tests.

It also fixes a race in the test due to the add callback being invoked in the run goroutine.

### CODE REVIEW

File: pool/endpoint_test.go [hunk 1]
```diff
@@ -91,7 +91,18 @@ func testEndpoint(t *testing.T) {
 		endpoint.run(ctx)
 		wg.Done()
 	}()
-	time.Sleep(time.Millisecond * 100)
+	sendToConnChanOrFatal := func(msg *connection) {
+		select {
+		case endpoint.connCh <- msg:
+		case <-ctx.Done():
+			t.Fatalf("unexpected enpoing shutdown")
```
Comment (jholdstock) [inline pool/endpoint_test.go:98]:
```suggestion
			t.Fatalf("unexpected endpoint shutdown")
```

File: pool/endpoint_test.go [hunk 2]
```diff
@@ -91,7 +91,18 @@ func testEndpoint(t *testing.T) {
...
+		}
+		select {
+		case <-msg.Done:
+		case <-ctx.Done():
+			t.Fatalf("unexpected enpoing shutdown")
```
Comment (jholdstock) [inline pool/endpoint_test.go:103]:
```suggestion
			t.Fatalf("unexpected endpoint shutdown")
```

File: pool/endpoint_test.go (hunk 1 above)
Comment (davecgh) [reply pool/endpoint_test.go]: Fixed.

File: pool/endpoint_test.go (hunk 2 above)
Comment (davecgh) [reply pool/endpoint_test.go]: Fixed.

**Review by jholdstock — APPROVED** [review]

**Pull request merged**